api/tests/Behat/bootstrap/v2/Common/ExpectedResultsTrait.php

```diff
@@ -6,83 +6,113 @@
-//            var_dump($section);
-//        }
-//        var_dump($expectedResultArrays);
+        if ($debug) {
+            $this->debugExpectedResultsDisplayed($sections, $summarySectionNumber, $expectedResults);
+        }
 
-        foreach ($sections[$sectionNumber] as $foundResultKey => $foundResults) {
-            $this->checkContainsExpectedResults($expectedResultArrays[$foundResultKey], $foundResults, $sectionNumber, $context);
+        foreach ($sections[$summarySectionNumber] as $foundResultKey => $foundResults) {
+            $this->checkContainsExpectedResults($expectedResults[$foundResultKey], $foundResults, $summarySectionNumber, $context);
         }
     }
 
-    private function getTextItems($items)
+    private function updateSummarySectionItemsFoundFromListRow($descriptionList)
+    {
+        $xpath = '//li';
+        $listItems = $descriptionList->findAll('xpath', $xpath);
+
+        if (count($listItems) > 0) {
+            $this->updateSummarySectionItemsFound($listItems);
+        } else {
+            $xpath = '//dt|//dd';
+            $descriptionDataItems = $descriptionList->findAll('xpath', $xpath);
+            $this->updateSummarySectionItemsFound($descriptionDataItems);
+        }
+    }
+
+    private function updateSummarySectionItemsFoundFromNoListRow($descriptionList)
+    {
+        $xpath = '//li';
+        $listItems = $descriptionList->findAll('xpath', $xpath);
+
+        if (count($listItems) > 0) {
+            $this->updateSummarySectionItemsFound($listItems);
+        } else {
+            $xpath = '//dt|//dd';
+            $descriptionDataItems = $descriptionList->findAll('xpath', $xpath);
+            $this->updateSummarySectionItemsFound($descriptionDataItems);
+        }
+    }
+
+    private function updateSummarySectionItemsFoundFromDescriptionList($descriptionList)
+    {
+        $xpath = "//div[contains(@class, 'govuk-summary-list__row')]";
+        $listSummaryRowItems = $descriptionList->findAll('xpath', $xpath);
+
+        if (count($listSummaryRowItems) > 0) {
+            $this->updateSummarySectionItemsFoundFromListRow($descriptionList);
+        } else {
+            $this->updateSummarySectionItemsFoundFromNoListRow($descriptionList);
+        }
+    }
+
+    private function updateSummarySectionItemsFoundFromTableBody($table)
+    {
+        $xpath = '//tr';
+        $tableRowItems = $table->findAll('xpath', $xpath);
+
+        if (count($tableRowItems) > 0) {
+            foreach ($tableRowItems as $tableRowItem) {
+                $xpath = '//td|//th';
+                $tableDataItems = $tableRowItem->findAll('xpath', $xpath);
+                $this->updateSummarySectionItemsFound($tableDataItems);
+            }
+        } else {
+            $xpath = '//td|//th';
+            $tableDataItems = $table->findAll('xpath', $xpath);
+            $this->updateSummarySectionItemsFound($tableDataItems);
+        }
+    }
+
+    private function updateSummarySectionItemsFound($items)
```

Comment:
We're never changing anything in `summarySectionItemsFound` in this function so maybe `addToSummarySectionItemsFound ` is clearer?

Reply:
good point. done